alexandria/core/views.py

```diff
@@ -139,6 +139,27 @@ def multi(self, request, **kwargs):
 
             return response
 
+    @action(methods=["post"], detail=True, url_path="generate-thumbnail")
+    def generate_thumbnail(self, request, pk=None):
+        if not settings.ALEXANDRIA_ENABLE_THUMBNAIL_GENERATION:
+            raise ValidationError(_("Thumbnail generation is not enabled."))
+
+        file = self.get_object()
+        if file.variant == models.File.THUMBNAIL:
+            raise ValidationError(
+                _("File is already a thumbnail, cannot generate thumbnail.")
+            )
+        elif file.renderings.all().count() > 0:
+            raise ValidationError(
+                _("File already has thumbnail, cannot generate multiple thumbnails.")
+            )
+
+        create_thumbnail(file)
```

Comment:
I wonder if we should check the return status of the `create_thumbnail()` call and generate a corresponding HTTP response.

When we're in a webhook from Minio, there's no recorvery so we don't care, but here, it's (presumably) a human-driven interaction where some success/failure state could be interesting.

Reply:
Returning 400 if it didnt succed